lib/seahorn/Smt/Yices2SolverImpl.cc

```diff
@@ -27,16 +27,24 @@ inline void yices_library_initialize(void){
   }
 }
 
-yices_solver_impl::yices_solver_impl(expr::ExprFactory &efac, solver_options opts)
-  : Solver(),
-    d_efac(efac) {
-  
+yices_solver_impl::yices_solver_impl(expr::ExprFactory &efac, const char *logic,
+                                     solver_options opts)
+    : Solver(), d_efac(efac) {
+
   yices_library_initialize();
   /* the yices configuration structure */
-  ctx_config_t *cfg = nullptr;
-  
+  ctx_config_t *cfg = yices_new_config();
+  if (logic != nullptr) {
+    int32_t res = yices_default_config_for_logic(cfg, logic);
```

Comment:
this is great, but the logic should be taken from `--horn-bmc-logic` option, not hard coded. That's why yices tests are failing right now.